CHANGELOG.md

```diff
@@ -99,16 +100,104 @@ Adding a new version? You'll need three changes:
  - [0.0.5](#005)
  - [0.0.4 and prior](#004-and-prior)
 
-## Unreleased
+## [3.4.0]
 
-### Deprecated
+> Release date: 2024-12-18
 
-- CRD type bindings under `/pkg` and clientsets under `/pkg/clientset` are deprecated
-  and will be removed in the next major release. Until then, they won't be updated.
-  They were migrated to a new dedicated [repository][kconf]. If you depend on them,
-  please update your dependencies to use the new repository.
+### Highlights
 
-[kconf]: https://github.com/kong/kubernetes-configuration
+- 🎂 New LTS version.
+- 🔒 Added Upstream TLS verification, both via `Service` annotations and Gateway
+  API `BackendTLSPolicy`. By using this feature, it is now possible to set up encrypted
+  communication between Kong and the backends.
+- 🎓 `KongCustomEntity` API has been promoted from beta to GA.
+- 🔎 Added `/debug/config/diff-report` diagnostic endpoint. It returns the latest
+  diff information for the controller's last configuration sync along with config
+  hash and sync timestamp metadata.
```

Comment:
It's probably worth mentioning it's only supported in DB-mode to not confuse users who would like to try it out.
```suggestion
- 🔎 Added `/debug/config/diff-report` diagnostic endpoint. It returns the latest
  diff information for the controller's last configuration sync along with config
  hash and sync timestamp metadata. It's only supported with Gateways running in DB mode.
```